### PR TITLE
Java: Refactor `java/static-initialization-vector` to use Models as Data

### PR DESCRIPTION
**Pull request by egregius313**
Refactors the `java/static-initialization-vector` query to use models-as-data models for it's sink and moves the definitions for `isAdditionalFlowStep` into models.

This makes it easier for other queries to use these definitions (this is beneficial for queries such as Weak Randomness).

### CODE REVIEW

File: java/ql/lib/semmle/code/java/security/StaticInitializationVectorQuery.qll [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import java
 import semmle.code.java.dataflow.TaintTracking
 import semmle.code.java.dataflow.DataFlow2
```
Comment (atorralba) [inline java/ql/lib/semmle/code/java/security/StaticInitializationVectorQuery.qll:5]: Drive-by change: we can just remove this import and replace `DataFlow2` with `DataFlow` in its only use below.

```suggestion
```

**Review by atorralba — COMMENTED** [review]
LGTM, I don't expect the previously query-only steps converted to global summaries to have that much of an impact on other queries. 1 minor drive-by change suggestion now that you're changing this query.

**Review by atorralba — APPROVED** [review]

**Review by owen-mc — APPROVED** [review]
Proxying atorralba 's review.
